### PR TITLE
Allow any ArrowReader implementation to be use for reading Arrow data #627

### PR DESCRIPTION
**Pull request by fb64**
ArrowFileReader part use specific method while ArrowStreamReader part rely only on ArrowReader interface methods. This modification just enables any ArrowReader implementation to be use for reading arrow data

### CODE REVIEW

File: dataframe-arrow/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/ArrowKtTest.kt [hunk 1]
```diff
@@ -29,21 +33,24 @@ import org.jetbrains.kotlinx.dataframe.api.columnOf
...
 import java.io.ByteArrayOutputStream
 import java.io.File
 import java.net.URL
 import java.nio.channels.Channels
+import java.sql.DriverManager
 import java.time.LocalDate
 import java.time.LocalDateTime
 import java.time.ZoneOffset
-import java.util.Locale
+import java.util.*
```
Comment (Jolanrensen) [inline dataframe-arrow/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/ArrowKtTest.kt:53]: We turn off * imports in the project. So could you do that for consistency too?

File: dataframe-arrow/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/ArrowKtTest.kt [hunk 2]
```diff
@@ -558,23 +565,26 @@ internal class ArrowKtTest {
         }
     }
 
-    @Test
-    fun testArrowReaderExtension() {
+    private fun expectedSimpleDataFrame(): AnyFrame{
```
Comment (Jolanrensen) [inline dataframe-arrow/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/ArrowKtTest.kt:568]: missed a space before "{"

Comment (fb64) [reply dataframe-arrow/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/ArrowKtTest.kt]: Fixed, weird thing that was not detected by linter during my local build

Comment (Jolanrensen) [reply dataframe-arrow/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/ArrowKtTest.kt]: Yes, our linter is a bit broken :/ https://github.com/Kotlin/dataframe/issues/364

**Comment by Jolanrensen** [comment]
CI agrees, I merged it, thanks!

**Pull request merged**